crates/re_viewer_context/src/time_control.rs

```diff
@@ -375,33 +375,39 @@ impl TimeControl {
 
     /// playback fps
     pub fn fps(&self) -> Option<f32> {
-        self.states.get(&self.timeline).map(|state| state.fps)
+        self.states.get(self.timeline()).map(|state| state.fps)
     }
 
     /// playback fps
     pub fn set_fps(&mut self, fps: f32) {
-        if let Some(state) = self.states.get_mut(&self.timeline) {
+        if let Some(state) = self.states.get_mut(self.timeline.get()) {
             state.fps = fps;
         }
     }
 
     /// Make sure the selected timeline is a valid one
     pub fn select_a_valid_timeline(&mut self, times_per_timeline: &TimesPerTimeline) {
-        for timeline in times_per_timeline.timelines() {
-            if &self.timeline == timeline {
-                return; // it's valid
+        fn is_timeline_valid(selected: &Timeline, times_per_timeline: &TimesPerTimeline) -> bool {
+            for timeline in times_per_timeline.timelines() {
+                if selected == timeline {
+                    return true; // it's valid
+                }
             }
+            false
         }
-        if let Some(timeline) = default_timeline(times_per_timeline.timelines()) {
-            self.timeline = *timeline;
-        } else {
-            self.timeline = Default::default();
+
+        // If the timeline is auto refresh it every frame, otherwise only pick a new one if invalid.
+        if self.timeline.is_auto() || !is_timeline_valid(self.timeline(), times_per_timeline) {
+            self.timeline = EditableAutoValue::Auto(
+                default_timeline(times_per_timeline.timelines()).map_or(Default::default(), |t| *t),
```

Comment:
The alphabetical order behavior is a bit unexpected (though I agree still a net improvement).

Since `times_per_timeline` persists with the EntityDb, internally tracking insertion order as part of `TimesPerTimeline::insert()` wouldn't be all that painful though.

Reply:
Hmm I think you're right. I'll try to go the extra step tomorrow

Reply:
poked it a bit, but changing this BTreeMap of timelines right now is a bit too annoying for the moment and then it gets less viable for a patch release